worker/firewaller/firewaller_test.go

```diff
@@ -834,6 +849,35 @@ func (s *InstanceModeSuite) TestRemoteRelationRequirerRoleConsumingSide(c *gc.C)
 	}
 }
 
+func (s *InstanceModeSuite) TestRemoteRelationWorkerError(c *gc.C) {
+	published := make(chan bool)
+	ingressRequired := true
+	apiErr := true
+	fw, ru := s.setupRemoteRelationRequirerRoleConsumingSide(c, published, &apiErr, &ingressRequired, testing.NewClock(time.Time{}))
+	defer statetesting.AssertKillAndWait(c, fw)
+
+	// Add a unit on the consuming app and have it enter the relation scope.
+	// This will trigger the firewaller to try and publish the changes.
+	err := ru.EnterScope(map[string]interface{}{})
+	c.Assert(err, jc.ErrorIsNil)
+
+	// We should not have published any ingress events yet - no changed published.
+	select {
+	case <-time.After(coretesting.ShortWait):
+	case <-published:
+		c.Fatal("unexpected ingress change to be published")
+	}
+
+	// Give the worker time to restart and try again.
+	apiErr = false
+	s.clock.(*testing.Clock).WaitAdvance(60*time.Second, coretesting.LongWait, 1)
```

Comment:
how is this working? AFAICS, you're not passing the test clock into the relationWorkerRunner constructor. working by coincidence?

Reply:
FFS, good pickup. Now sure how it worked, but with the fix to pass the clock to the runner, I confirmed that commenting out the WaitAdvance causes the test to fail.